fork/fork.go

```diff
@@ -19,37 +19,47 @@ package fork
 import (
 	"errors"
 	"math/big"
-	"sync"
 
+	"github.com/klaytn/klaytn/log"
 	"github.com/klaytn/klaytn/params"
 )
 
 var (
 	// hardForkBlockNumberConfig contains only hardFork block number
 	hardForkBlockNumberConfig *params.ChainConfig
-	// once limits the usage of `SetHardForkBlockNumberConfig`
-	once sync.Once
+
+	logger = log.NewModuleLogger(log.FORK)
 )
 
 // Rules returns the hard fork information
 // CAUTIOUS: Use it when chainConfig value is not reachable
-func Rules(blockNumber *big.Int) (*params.Rules, error) {
+func Rules(blockNumber *big.Int) *params.Rules {
 	if hardForkBlockNumberConfig == nil {
-		return nil, errors.New("hardForkBlockNumberConfig variable is not initialized")
+		logger.Crit("fork.Rules should never be called before hardForkBlockNumberConfig initialization.")
 	}
 	rules := hardForkBlockNumberConfig.Rules(blockNumber)
-	return &rules, nil
+	return &rules
 }
 
 // SetHardForkBlockNumberConfig sets values in HardForkConfig if it is not nil.
 // CAUTIOUS: Calling this function is can be dangerous, so avoid using it except tests
 func SetHardForkBlockNumberConfig(h *params.ChainConfig) error {
+	// ensure that the allocation is done only when h is not nil
 	if h == nil {
 		return errors.New("hardForkBlockNumberConfig cannot be initialized as nil")
 	}
-	// ensure that this statement is executed only once
-	once.Do(func() {
-		hardForkBlockNumberConfig = h
-	})
+	hardForkBlockNumberConfig = h
 	return nil
 }
+
+// GetHardForkBlockNumberConfig returns hardForkBlockNumberConfig
+// Use only in tests
+func GetHardForkBlockNumberConfig() *params.ChainConfig {
+	return hardForkBlockNumberConfig
+}
+
+// ResetHardForkBlockNumberConfig sets nil
+// Use only in tests
+func ResetHardForkBlockNumberConfig() {
```

Comment:
I think "Clear~" is more proper than "Reset~" for this function name.

Reply:
Thanks, aidan.. Changed to Clear in new change https://github.com/klaytn/klaytn/pull/997/commits/cc27ea34d041de3bbef823c5ab78365fa73d3b37!

Reply:
It is still `Reset` to me.. Could you please check? @yoomee1313

Reply:
It is still Reset to me.. Could you please check?

@ehnuje oh! I noticed that change is missing too.. i fixed it, sorry for confusion!